core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1409,27 +1409,27 @@ else if (res.operationResult == RETRY) {
             this.internalOakMap = internalOakMap;
         }
 
-        public Map.Entry<OakRBuffer, OakRBuffer> next() {
+        public Map.Entry<OakDetachedBuffer, OakDetachedBuffer> next() {
             IterItem myItem = advance(true);
             long keyReference = myItem.keyReference;
             long valueReference = myItem.valueReference;
             int version = myItem.valueVersion;
             return new AbstractMap.SimpleImmutableEntry<>(setKeyReference(keyReference,
-                    new OakRKeyBuffer(memoryManager, KEY_HEADER_SIZE)), new OakRValueBuffer(valueReference,
+                    new OakDetachedReadKeyBuffer(memoryManager, KEY_HEADER_SIZE)), new OakDetachedReadValueBuffer(valueReference,
```

Comment:
Is the difference between "OakDetachedReadValueBuffer" and "OakDetachedReadKeyBuffer" when applied on value, only in the fact that the first provides synchronization and the second not?

Reply:
They both provide some kind of synchronization method, but a different one.
Key is synchronized via a hazard pointer, and value via a lock.